File: data/src/main/java/com/ykb/app/cryptotrader/data/model/ErrorLog.java
package com.ykb.app.cryptotrader.data.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Lob;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.io.PrintWriter;
import java.io.StringWriter;

@Entity
@Table(name = "ERROR_LOG")
@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ErrorLog extends BaseIdEntity {

    @Column(columnDefinition = "VARCHAR(2048)", nullable = false)
    private String message;

    @Lob
    @Column(name = "STACK_TRACE")
    private String stackTrace;

    public ErrorLog(String msg) {
        this(msg, null);
    }

    public ErrorLog(Exception e) {
        this(null, e);
    }

    public ErrorLog(String msg, Exception e) {
        message = msg == null ? e.getMessage() : msg;
        stackTrace = getStackTraceAsString(e);
    }

    private String getStackTraceAsString(Exception e) {
        StringWriter sw = new StringWriter();
        e.printStackTrace(new PrintWriter(sw));
        return sw.toString();
    }

}

File: library/utils/src/main/kotlin/com/ykb/app/cryptotrader/utils/logger/data/ErrorLog.kt
package com.ykb.app.cryptotrader.utils.logger.data

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.Temporal
import jakarta.persistence.TemporalType
import java.util.Date

@Entity
@Table(name = "ERROR_LOG")
class ErrorLog(
    ex: Throwable,
    @Column(columnDefinition = "VARCHAR(256)", nullable = true)
    @Enumerated(EnumType.STRING)
    private val description: String?
) {

    constructor(ex: Throwable): this(ex, ex.message)

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Int? = null

    @Column(name = "CREATION_TIME")
    @Temporal(TemporalType.TIMESTAMP)
    private val creationTime = Date()

    @Column(columnDefinition = "TEXT")
    private val stacktrace: String = ex.stackTraceToString()
}